packages/state-importer/src/main.js

```diff
@@ -0,0 +1,240 @@
+/* @flow */
+import Promise from 'bluebird'
+import { createClient } from '@commercetools/sdk-client'
+import { createRequestBuilder } from '@commercetools/api-request-builder'
+import {
+  createAuthMiddlewareForClientCredentialsFlow,
+  createAuthMiddlewareWithExistingToken,
+} from '@commercetools/sdk-middleware-auth'
+import { createHttpMiddleware } from '@commercetools/sdk-middleware-http'
+import { createUserAgentMiddleware } from '@commercetools/sdk-middleware-user-agent'
+import { createSyncStates } from '@commercetools/sync-actions'
+import pkg from '../package.json'
+import type {
+  ApiConfigOptions,
+  ConstructorOptions,
+  LoggerOptions,
+  StateData,
+  Summary,
+} from '../../../types/state'
+import type { Client, SyncAction } from '../../../types/sdk'
+
+class StateImportError extends Error {
+  error: any
+  summary: string
+  message: string
+
+  constructor(
+    message: string,
+    summary: string = 'No summary provided.',
+    error: any = null
+  ) {
+    super(message)
+
+    this.message = message
+    this.summary = summary
+    this.error = error
+  }
+}
+
+export default class StateImport {
+  // Set flowtype annotations
+  accessToken: string
+  continueOnProblems: boolean
+  client: Client
+  apiConfig: ApiConfigOptions
+  logger: LoggerOptions
+  _summary: Summary
+  syncStates: SyncAction
+
+  constructor(options: ConstructorOptions, logger: LoggerOptions) {
+    if (!options.apiConfig)
+      throw new StateImportError(
+        'The constructor must be passed an `apiConfig` object'
+      )
+
+    this.apiConfig = options.apiConfig
+    this.accessToken = options.accessToken
+    this.continueOnProblems = options.continueOnProblems || false
+    this.client = createClient({
+      middlewares: [
+        createAuthMiddlewareWithExistingToken(
+          this.accessToken ? `Bearer ${this.accessToken}` : ''
+        ),
+        createAuthMiddlewareForClientCredentialsFlow(this.apiConfig),
+        createUserAgentMiddleware({
+          libraryName: pkg.name,
+          libraryVersion: pkg.version,
+        }),
+        createHttpMiddleware({ host: this.apiConfig.apiUrl }),
+      ],
+    })
+
+    this.syncStates = createSyncStates()
+
+    this.logger = logger || {
+      error: () => {},
+      warn: () => {},
+      info: () => {},
+      debug: () => {},
+    }
```

Comment:
Somewhat feels like eventually some time we want a bit smarter setup for logging. With levels and hooks kind of.

Reply:
This is something that can be looked into. But changing this here will break compatibility with the [node-importer](https://github.com/sphereio/sphere-node-cli)